src/sqlfluff/core/config.py

```diff
@@ -260,14 +260,24 @@ def load_config_at_path(self, path: str) -> dict:
         self._config_cache[str(path)] = configs
         return configs
 
-    def load_user_appdir_config(self) -> dict:
+    def load_user_appdir_config(self) -> Tuple[Optional[str], dict]:
         """Load the config from the user's OS specific appdir config directory."""
         appname = "sqlfluff"
         appauthor = "sqlfluff"
-        user_config_dir_path = appdirs.user_config_dir(appname, appauthor)
+
+        # On Mac OSX follow Linux XDG base dirs
+        # https://github.com/sqlfluff/sqlfluff/issues/889
+        user_config_dir_path = os.path.expanduser("~/.config/sqlfluff")
+        if appdirs.system == "darwin":
```

Comment:
This looks a little tricky, switching the OS. Is there a known issue with `appdirs` requiring this technique?

Reply:
the context is here: https://github.com/ActiveState/appdirs/issues/78#issuecomment-366947045